### DISCUSSION TITLE
Environment Variables in NextJS

### DISCUSSION THREAD

**Discussion by MarcusHSmith** [vercel/community#213] [Help] [answered]
Hello,

I'm having some issues with Environment Variables and want to sanity check myself.

I have a NextJS application deployed with Vercel. I have a Mailchimp signup form in my NextJS app in the `/api` directory. This references Environment Variables store in Vercel. The request is failing.

_I think_ this is because the [docs](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser) say that Environment Variables can only be used in Node.JS. If I understand the solution correct, I could update my Environment Variables to be `NEXT_PUBLIC_MAILCHIMP_API_KEY` in both the NextJS code and Vercel. I could also put this request in my Express backend and have my client app trigger this in my backend.

Am I understanding the problem correctly?

**Comment by Pieparker** [comment, marked as answer]
Sounds plausible; it'll depend how and where it's failing. Though, to help, you can [give this guidance a follow](https://leerob.io/blog/mailchimp-next-js). Lee knows his stuff!
